hazelcast/src/main/java/com/hazelcast/map/impl/recordstore/StorageImpl.java

```diff
@@ -92,7 +93,8 @@ public void put(Data key, R record) {
     public void updateRecordValue(Data key, R record, Object value) {
         updateCostEstimate(-entryCostEstimator.calculateValueCost(record));
 
-        recordFactory.setValue(record, value);
+        record.setValue(inMemoryFormat == BINARY
+                ? ss.toData(value) : ss.toObject(value));
```

Comment:
Needs to use the map partitioning strategy.

Reply:
do we need it for values?

Reply:
True, we don't.